### ISSUE TITLE
Method_DeviceMethodRespLinkDropRecovery_AmqpWs failed on CI run

### ISSUE THREAD

**Issue by CIPop** [Azure/azure-iot-sdk-csharp#194]
Multiple tests failing intermittently with DeviceNotFoundException:

```
Microsoft.Azure.Devices.E2ETests.MethodE2ETests.Method_DeviceMethodRespLinkDropRecovery_Amqp
Microsoft.Azure.Devices.E2ETests.MethodE2ETests.Method_DeviceMethodRespLinkDropRecovery_AmqpWs
Microsoft.Azure.Devices.E2ETests.MethodE2ETests.Method_DeviceMethodReqLinkDropRecovery_AmqpWs
```

```Test method Microsoft.Azure.Devices.E2ETests.MethodE2ETests.Method_DeviceMethodRespLinkDropRecovery_AmqpWs threw exception: 
Microsoft.Azure.Devices.Common.Exceptions.DeviceNotFoundException: Device {"Message":"{\"errorCode\":404103,\"trackingId\":\"6106d4daadfb4c0c8754d0b366647244-G:2-TimeStamp:10/18/2017 00:08:58\",\"message\":\"Timed out waiting for device to connect.\",\"info\":{\"timeout\":\"00:00:00\"},\"timestampUtc\":\"2017-10-18T00:08:58.2412966Z\"}","ExceptionMessage":""} not registered

Microsoft.Azure.Devices.HttpClientHelper.<ExecuteAsync>d__34.MoveNext() in D:\jenkins\workspace\windows-csharp-wip-test\common\src\service\HttpClientHelper.cs: line 769
System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
Microsoft.Azure.Devices.HttpClientHelper.<PostAsync>d__25`2.MoveNext() in D:\jenkins\workspace\windows-csharp-wip-test\common\src\service\HttpClientHelper.cs: line 446
System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
Microsoft.Azure.Devices.E2ETests.MethodE2ETests.<ServiceSendMethodAndVerifyResponse>d__38.MoveNext() in D:\jenkins\workspace\windows-csharp-wip-test\e2e\Microsoft.Azure.Devices.E2ETests\MethodE2ETests.cs: line 295
System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task task)
Microsoft.Azure.Devices.E2ETests.MethodE2ETests.<SendMethodAndRespondRecovery>d__41.MoveNext() in D:\jenkins\workspace\windows-csharp-wip-test\e2e\Microsoft.Azure.Devices.E2ETests\MethodE2ETests.cs: line 432
System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
Microsoft.Azure.Devices.E2ETests.MethodE2ETests.<Method_DeviceMethodRespLinkDropRecovery_AmqpWs>d__35.MoveNext() in D:\jenkins\workspace\windows-csharp-wip-test\e2e\Microsoft.Azure.Devices.E2ETests\MethodE2ETests.cs: line 258
System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
System.Runtime.CompilerServices.TaskAwaiter.GetResult()
```

/cc @jasmineymlo @ryan-z-johnson @tameraw

**Comment by CIPop** [comment]
Two more failed today.

**Comment by jenoola** [comment]
Fixed in 1.20.0

**Comment by az-iot-builder-01** [comment]
@CIPop, @jenoola, thank you for your contribution to our open-sourced project! Please help us improve by filling out this 2-minute [customer satisfaction survey](https://forms.office.com/Pages/ResponsePage.aspx?id=v4j5cvGGr0GRqy180BHbRx8jpG79OudPqY-BojcKjkJUMFFBQjgxNkw5ODJIRjBWTzI0Nk85R0pUQiQlQCN0PWcu)
